When publishing Protobuf source JARs, publish the dependencies.

SamirTalwar-DA opened the pull request:
Previously, each Protobuf source JAR was considered independent, and did not require dependencies. This meant that the kvutils Protobuf sources were published without the associated DAML-LF sources, making them useless.

This change adds the source JARs as runtime dependencies to ensure that when publishing a Protobuf source JAR, the release process will fail if the dependencies are not also published.

### Changelog

- **[Ledger API]** We now publish Protobuf sources in JARs for DAML-LF types to Maven Central, with the group "com.daml". The artifact names follow the format "<name>_proto_jar".

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

fabiotudone-da commented on bazel_tools/proto.bzl:
```diff
@@ -220,9 +220,11 @@ def proto_jars(
         resources = srcs,
         resource_strip_prefix = "%s/%s/" % (native.package_name(), strip_import_prefix),
         tags = _maven_tags(maven_group, maven_artifact_prefix, maven_artifact_proto_suffix),
+        visibility = visibility,
```
What does that do?

SamirTalwar-DA replied:
It makes these JARs visible outside the current _BUILD.bazel_ file calling `proto_jars`, which means that we can actually establish the dependencies. For example, the kvutils proto JAR needs to be able to see the DAML-LF transaction proto JAR in order to depend upon it.

aherrmann-da replied:
@SamirTalwar-DA Did it fail without this? IIRC the release script collects these targets via `bazel build` instead of dependencies declared in BUILD files. So, the visibility shouldn't matter.

aherrmann-da replied:
Oh, it's added to `runtime_deps`. Nvm, makes sense.

SamirTalwar-DA replied:
@aherrmann-da: Yes; see the above comment. Bazel itself has to resolve the `runtime_deps`, so they need to see each other.

fabiotudone-da reviewed (approved):
@SamirTalwar-DA LGTM, were you able to test that publishing fails if the deps are not published?

SamirTalwar-DA commented:
@fabiotudone-da: Yes, I added the `runtime_deps` first and ensured that `daml-sdk-head` failed with an error listing the missing dependencies. Then I just added the dependencies listed to _artifacts.yaml_.
